### DISCUSSION TITLE
How the remote client simply waits for rabbitmq-server to be ready

### DISCUSSION THREAD

**Discussion by lizhifengones** [rabbitmq/rabbitmq-server#4280] [Other] [answered]
I see that rmq has added a lot of cool health checks (https://github.com/rabbitmq/rabbitmq-management/issues/844).

I am a rookie of rabbitmq. When I have some microservices in k8s that depend on rabbitmq, I will consider using init-container to wait for rabbitmq, but I have not found a good way.

**Comment by michaelklishin** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(

**Comment by michaelklishin** [comment, marked as answer]
Applications that need to connect to RabbitMQ are not expected to use health checks for that. They should handle connection failures and use [a list of endpoints](https://rabbitmq.com/api-guide.html#endpoints-list) that many clients support. Health checks are for monitoring tools, not services that communicate over RabbitMQ.

**Reply by lizhifengones** [reply]
Thanks for your reply, your suggestion is correct. Another question is that the health check API has been exposed, right? We no longer need to be able to query the status of the rmq node locally, right?

**Reply by michaelklishin** [reply]
I don't understand the question I'm afraid. What does "locally" mean? What did you query it for? The only thing that's changed
around health checks is the deprecation of a legacy intrusive check and a [set of new ones](https://rabbitmq.com/monitoring.html#health-checks). The way health checks are used is pretty much the satgitme.

Not sure where/if the HTTP API endpoints are documented but [here they are](https://github.com/rabbitmq/rabbitmq-server/blob/master/deps/rabbitmq_management/src/rabbit_mgmt_dispatcher.erl#L169-L177) in the dispatcher code.
